versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/DatabaseAdapterConfig.java

```diff
@@ -43,11 +43,9 @@
    */
   @Value.Default
   default String getKeyPrefix() {
-    return "";
+    return DEFAULT_KEY_PREFIX;
   }
 
-  DatabaseAdapterConfig<T> withKeyPrefix(String keyPrefix);
```

Comment:
Isn't there a way to tell Smallrye-Config/Quarkus to ignore these methods?

Reply:
not sure... if we kept them how would quarkus-based sub-classes implement them? :thinking:

Reply:
Custom code that wants to create configs from scratch can use those "with" and other builder methods from the `Immutable` implementations, I guess.

Reply:
I've added `with...` methods mostly to ease configuration using those `.configurate(c -> c)` methods. Seems, that this is mostly no longer necessary, so I'm fine with removing those.